### PR TITLE
HOSTEDCP-1044: Add NodePool Size/Replicas to Telemetry

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-1044](https://issues.redhat.com/browse/HOSTEDCP-1044)

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: cmd/install/assets/recordingrules/hypershift.yaml [hunk 1]
```diff
@@ -83,4 +83,10 @@ groups:
     expr: max by(platform) (hypershift_hostedclusters)
     
   - record: platform:hypershift_nodepools:max
-    expr: max by(platform) (hypershift_nodepools)
\ No newline at end of file
+    expr: max by(platform) (hypershift_nodepools)
+
+  - record: hypershift:nodepools:size
+    expr: max by(name, platform) (hypershift_nodepools_size)
```
Comment (simonpasquier) [inline cmd/install/assets/recordingrules/hypershift.yaml:89]: "name_platform:hypershift_nodepools_size:max" would be a more descriptive name. What's the expected cardinality for the name and platform labels?

Comment (muraee) [reply cmd/install/assets/recordingrules/hypershift.yaml:89]: name and platform labels identify a unique nodepool, I can't tell what the expected cardinality would be. It boils down to how many nodepools are created in total in a management cluster.
assuming an average of 3 Nodepools per HostedCluster, and the maximum number of 80 HostedClusters we end up with 240 nodepools/timeseries.

Comment (simonpasquier) [reply cmd/install/assets/recordingrules/hypershift.yaml:89]: 240 series is a lot and needs an explicit approval (see https://rhobs-handbook.netlify.app/products/openshiftmonitoring/telemetry.md/#request-approval).

Comment (simonpasquier) [reply cmd/install/assets/recordingrules/hypershift.yaml:89]: cc @jan--f

Comment (jan--f) [reply cmd/install/assets/recordingrules/hypershift.yaml:89]: And I assume the same is true for `name_platform:hypershift_nodepools_available_replicas:max`? That is  quite a lot of additional time series. 
Unless you really need this level of detail per node pool, I'd recommend further aggregation.

Comment (muraee) [reply cmd/install/assets/recordingrules/hypershift.yaml]: changed to aggregate per HostedCluster, the maximum cardinality would between 80-100

Comment (simonpasquier) [reply cmd/install/assets/recordingrules/hypershift.yaml]: This is still a lot. Can't you do min/max/median + a few percentiles?

Comment (muraee) [reply cmd/install/assets/recordingrules/hypershift.yaml]: I am not sure if min/max/median + a few percentiles would give us any real insights anymore.
cc @zanetworker

**Comment by muraee** [comment]
@simonpasquier replaced `_id` with `cluster_name` could you please take another look.

**Review by simonpasquier — COMMENTED** [review]
IIUC `cluster_name` is the name of the HCP cluster and it's a namespaced resource. Don't you need to include the "namespace" label in the sum aggregation?

File: cmd/install/assets/recordingrules/hypershift.yaml (hunk 1 above)
Comment (zanetworker) [reply cmd/install/assets/recordingrules/hypershift.yaml]: @simonpasquier that won't provide meaningful data if we are trying to understand how one customer is using the feature. We can always min/max/... outside of the metric.

**Comment by bryan-cox** [comment]
/retest

For konflux

**Pull request merged**